tests/issues/GH3051.test.ts

```diff
@@ -0,0 +1,66 @@
+import { BigIntType, Collection, Entity, ManyToOne, MikroORM, OneToMany, PrimaryKey, Property } from '@mikro-orm/core';
+
+@Entity()
+class Author {
+
+  @PrimaryKey({ type: BigIntType })
+  id?: string;
+
+  @Property()
+  name!: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-use-before-define
+  @OneToMany({ entity: () => Book, mappedBy: (book: Book) => book.author, orphanRemoval: true })
+  books: Collection<Book> = new Collection<Book>(this);
+
+}
+
+@Entity()
+class Book {
+
+  @PrimaryKey({ type: BigIntType })
+  id?: string;
+
+  @Property()
+  title!: string;
+
+  @ManyToOne({ entity: () => Author })
+  author!: Author;
+
+}
+
+describe('GH issue 3051', () => {
+  let orm: MikroORM;
+
+  beforeEach(async () => {
+    orm = await MikroORM.init({
+      entities: [Author, Book],
+      dbName: ':memory:',
+      type: 'sqlite',
+    });
+
+    await orm.getSchemaGenerator().createSchema();
+    const author = orm.em.create(Author, {
+      name: 'Jon Snow',
+      books: [{ title: 'b1' }, { title: 'b2' }, { title: 'b3' }],
+    });
+    await orm.em.fork().persistAndFlush(author);
```

Comment:
this feels wrong, your `em.create` call will be creating references for the books, they will stay in the identity map. 
```suggestion
    const author = orm.em.create(Author, {
      name: 'Jon Snow',
      books: [{ title: 'b1' }, { title: 'b2' }, { title: 'b3' }],
    });
    await orm.em.persistAndFlush(author);
    orm.em.clear();
```